src/NuGet.Core/NuGet.Frameworks/FrameworkReducer.cs

```diff
@@ -52,11 +52,11 @@ public NuGetFramework GetNearest(NuGetFramework framework, IEnumerable<NuGetFram
         {
             var nearest = GetNearestInternal(framework, possibleFrameworks);
 
-            var fallbackFramework = framework as FallbackFramework;
-
-            if (fallbackFramework != null)
+            if (nearest == null || nearest.IsAny)
```

Comment:
Combine the PTF/ATF checks and loops

1. Check if framework is ATF, read fallbacks
1. If not ATF check if PTF, read fallbacks
1. If fallbacks exist loop through them and check compat (doesn't matter where they came from)